quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/TransactionalResponseInterceptor.java

```diff
@@ -34,13 +35,21 @@
 @ConstrainedTo(RuntimeType.SERVER)
 @Priority(10000)
 public class TransactionalResponseInterceptor implements WriterInterceptor, TransactionalSessionHandler {
+
     @Override
-    public void aroundWriteTo(WriterInterceptorContext context) throws IOException, WebApplicationException {
+    public void aroundWriteTo(WriterInterceptorContext context) {
         KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
+        OutputStream outputStream = context.getOutputStream();
 
-        try {
+        // see https://github.com/keycloak/keycloak/issues/20451
+        try (ByteArrayOutputStream buffer = new ByteArrayOutputStream(8192)) {
+            context.setOutputStream(buffer);
             context.proceed();
+            outputStream.write(buffer.toByteArray());
```

Comment:
Using a buffered output stream should cause the same error we are trying to fix because bytes might be flushed to the underlying output stream before processing all elements in the stream.

Reply:
Materializing the response in memory increases the memory footprint for all the responses. To conserve memory, the output stream should rather be kept, perhaps with some reasonable buffer that would be cleared up if an exception is thrown.